File: golang/010.go
package main

import "fmt"

func bubbleSort(arr []int)[]int{
	for range arr{
		for i := range arr{
		if i+1 < len(arr){
			if arr[i] > arr[i+1]{
				arr[i], arr[i+1] = arr[i+1], arr[i]
			}
		}
	}}

	return arr
}

func recursiveCallBubbleSort(arr []int, pointer, iteration int) []int {
	//pointer reset
	if pointer == len(arr){
		pointer = 0
		iteration++
	}
	//base case
	if iteration == len(arr){
		return arr
	}
	if pointer + 1 < len(arr) && arr[pointer] > arr[pointer + 1]{
		arr[pointer], arr[pointer + 1] = arr[pointer + 1], arr[pointer]
	}
	return recursiveCallBubbleSort(arr, pointer+1, iteration)
}

func main(){
	fmt.Println(recursiveCallBubbleSort([]int{3,4,5,6,7,2,223,4,4,2},0,0))
}

File: python/010.py
def bubblesort(arr):
    for x in range(0,len(arr)):
        for x in range(0,len(arr)):
            if x + 1 < len(arr) and arr[x] > arr[x+1]:
                arr[x], arr[x+1] = arr[x+1], arr[x]
    return arr

print(bubblesort([2,3,5,6,7,2,3,4,2,4,453,5,4]))